pkg/kubelet/dockertools/manager.go

```diff
@@ -1071,70 +1096,86 @@ func (dm *DockerManager) KillPod(pod kubecontainer.Pod) error {
 	return nil
 }
 
-// KillContainerInPod kills a container in the pod.
-func (dm *DockerManager) KillContainerInPod(container api.Container, pod *api.Pod) error {
-	// Locate the container.
-	pods, err := dm.GetPods(false)
-	if err != nil {
-		return err
-	}
-	targetPod := kubecontainer.Pods(pods).FindPod(kubecontainer.GetPodFullName(pod), pod.UID)
-	targetContainer := targetPod.FindContainerByName(container.Name)
-	if targetContainer == nil {
-		return fmt.Errorf("unable to find container %q in pod %q", container.Name, targetPod.Name)
-	}
-	return dm.killContainer(targetContainer.ID)
-}
+// KillContainerInPod kills a container in the pod. It must be passed either a container ID or a container and pod,
+// and will attempt to lookup the other information if missing.
+func (dm *DockerManager) KillContainerInPod(containerID types.UID, container *api.Container, pod *api.Pod) error {
```

Comment:
Better yes.  I was concerned about the existing duplication (I looked at removing it) but more concerned about breaking something accidentally.

> On Jun 3, 2015, at 2:00 PM, Victor Marmol <notifications@github.com> wrote:
> 
> In pkg/kubelet/dockertools/manager.go:
> 
> > -func (dm *DockerManager) KillContainerInPod(container api.Container, pod *api.Pod) error {
> > -	// Locate the container.
> > -	pods, err := dm.GetPods(false)
> > -	if err != nil {
> > -		return err
> > -	}
> > -	targetPod := kubecontainer.Pods(pods).FindPod(kubecontainer.GetPodFullName(pod), pod.UID)
> > -	targetContainer := targetPod.FindContainerByName(container.Name)
> > -	if targetContainer == nil {
> > -		return fmt.Errorf("unable to find container %q in pod %q", container.Name, targetPod.Name)
> > -	}
> > -	return dm.killContainer(targetContainer.ID)
> > -}
> > +// KillContainerInPod kills a container in the pod. It must be passed either a container ID or a container and pod,
> > +// and will attempt to lookup the other information if missing.
> > +func (dm *DockerManager) KillContainerInPod(containerID types.UID, container *api.Container, pod *api.Pod) error {
> WDYT: Two different functions? KillContainerByID() and KillContainer()
> 
> —
> Reply to this email directly or view it on GitHub.
>